### DISCUSSION TITLE
Evaluate effects of Sturm snow conductivity changes

### DISCUSSION THREAD

**Discussion by wwieder** [NCAR/LMWG_dev#10] [General]
In #4 we implemented the Sturm snow conductivity changes suggested by [AdrienDams](https://github.com/AdrienDams) in [CTSM #2148](https://github.com/ESCOMP/CTSM/pull/2148).  As expected, this has big changes on winter soil temperatures in simulations by @olyson discussed in #3. **How do we independently evaluate the impact of these changes?**

![image](https://github.com/NCAR/LMWG_dev/assets/8031012/9d2b4b81-ec62-4bc5-a34b-99c22508b3f9)

**Comment by wwieder** [comment]
NEON simulations at TOOL that are forced with tower meteorology also have cold winter temperatures for the years that were able to run the tower.  Measurements show mean soil temperature (n=5) from sensors located ~5-6 cm depth & are compared with CLM results from simulations using the default and Sturm conductivity. 

I have no idea if soil moisture or snowpack is behaving sensibly in the model.  Given how quickly soil in the simulation freeze, I'd suspect they are too dry. Active layer thickness is ~3m in the control simulations simulations (and deeper with Sturm!), which seems like pretty deep permafrost for the North Slope of Alaska.  Nor have I carefully evaluated QC flags from NEON data, but there could be more to investigate with snowy NEON sites, especially in AK.  

![image](https://github.com/NCAR/LMWG_dev/assets/8031012/b248188b-d143-4b0f-b744-ca235ee23462)

**Comment by wwieder** [comment]
Switching to Sturm results in a pretty large loss of discontinuous permafrost zone, here in an F-case
![image](https://github.com/NCAR/LMWG_dev/assets/8031012/46962293-a049-44ee-81ec-4f6493168cb5)
And also for a GSWP3 forced run
![image](https://github.com/NCAR/LMWG_dev/assets/8031012/726f5234-8c20-4563-9d84-0eec6fb65efa)

**Reply by AdrienDams** [reply]
@wwieder it sad we didn't speak about that more yesterday. Looking to your graphs motivate me to look again why we don't observe a deeper ALT in our run. I am really sorry to say that the Sturm ALT maps I showed yesterday were wrong. As I told you, we used our own way to compute ALT and doing the period average messed up our fill values.

I fixed it and I have run the evaluation again and we have similar results than yours, if not worse, for our stand-alone case. Sturm run does show a significant loss of discontinuous permafrost.

This is ALT difference between Sturm and Control run for full period (the non-permafrost areas in Sturm are in gray now)

![ALT period diff 57_DOM02_051-57_DOM02_001](https://github.com/NCAR/LMWG_dev/assets/24732655/9ce75e8e-3e0f-46ec-ba3b-6ae20c95e66b)

And this is compared to ESA-CCI

![ALT_control_sturm](https://github.com/NCAR/LMWG_dev/assets/24732655/44dabb18-a098-4b43-953c-d714d7e0af2e)

We still think it works well where it is supposed to work. We need a way to distinguish tundra and mountain areas, then it will be ok. But I would agree Sturm might look less promising now.

**Reply by wwieder** [reply]
Thanks for posting these @AdrienDams .  Does anyone know if a data product that would help us visualize observational estimates of permafrost extent?  For example, I know [ILAMB](https://webext.cgd.ucar.edu/BHIST/b.e21.BHIST.f09_g17.CMIP6-historical.001/lnd/_build_CESM2_1G/HydrologyCycle/Permafrost/NSIDC/NSIDC.html#AllModels) uses an NSIDC product, are these boundaries simple enough to overlay on plots here?  Alternatively, maybe it's worth running results from some of these simulations through ILAMB, although I guess that would require a full historical simulation.

**Reply by AdrienDams** [reply]
FYI, that's how our permafrost extent look like compared to ESA-CCI

![PFR_control_sturm](https://github.com/NCAR/LMWG_dev/assets/24732655/a6b8968b-f23a-4fff-a5ad-75edb34c6214)

**Comment by wwieder** [comment]
Coupled simulations really amplify climate effects of the snow thermal conductivity schemes!  
Here are differences (Sturm - Jordan) in summer (JJA) air temperature from the GSWP3-forced I case
![image](https://github.com/NCAR/LMWG_dev/assets/8031012/3bc76a94-afaf-4f21-9c3e-76d0751d6c2d)
vs. differences in from the coupled (F-case)
![image](https://github.com/NCAR/LMWG_dev/assets/8031012/90da5fa7-47f6-4613-8a5e-5431f7d70d9e)

**Comment by olyson** [comment]
The newest version of ILAMB (2.7) doesn't seem to use NSIDC anymore.  Instead, there are a couple of other products that are listed as Brown2002 and Obu2018.  Here are the data information pages on those:

https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a02cPPEn08ctsm51d030_1deg_GSWP3V1_hist/lnd/_build_ILAMB2_7_CLM50CTSM51BGC/HydrologyCycle/Permafrost/Brown2002/Brown2002.html#DataInformation

https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a02cPPEn08ctsm51d030_1deg_GSWP3V1_hist/lnd/_build_ILAMB2_7_CLM50CTSM51BGC/HydrologyCycle/Permafrost/Obu2018/Obu2018.html#DataInformation

I agree that we should do a historical at some point with Sturm.

**Comment by olyson** [comment]
Some standard and ILAMB diagnostics from the historical SturmSnowtk and control runs (see https://github.com/NCAR/LMWG_dev/issues/30 for a description)

Standard diagnostics:(https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_SturmSnowtk_hist/lnd/ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_SturmSnowtk_hist.1995_2014-ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_hist.1995_2014/setsIndex.html)
ILAMB diagnostics:
https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_SturmSnowtk_hist/lnd/_build_ILAMB2_7_CTSM51Sturm/
NOTE: The ILAMB permafrost diagnostics for Sturm throws an error (the control is fine), so needs to be investigated.  In the meantime some plots of ATLMAX (1995-2014; SturmSnowtk, Control, SturmSnowtk-Control; top to bottom)
![ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_SturmSnowtk_hist ANN ALTMAX polar](https://github.com/NCAR/LMWG_dev/assets/18671367/eb3944fb-febe-42d6-91d7-15701a511e6f)
![ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_hist ANN ALTMAX polar](https://github.com/NCAR/LMWG_dev/assets/18671367/f8684f18-cfb7-42b2-95c9-f60eefd220bb)
![SturmSnowtk_1995-2014-Control_1995-2014 ANN ALTMAX diff polar](https://github.com/NCAR/LMWG_dev/assets/18671367/377d7148-07af-40d2-85b5-09e9f27f320f)

**Reply by wwieder** [reply]
Thanks for posting these, Keith.  Even here the diagnostics are helpful: we seem to have low permafrost extent,  based on the Brown 2002 estimates in the control case (red areas show missing permafrost in the model.   ![image](https://github.com/NCAR/LMWG_dev/assets/8031012/bf1f547b-56d9-4ad0-b8d8-cdc6cddf2a05)

This maybe isn't quite as bad for [Obu 2018 dataset](https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_SturmSnowtk_hist/lnd/_build_ILAMB2_7_CTSM51Sturm/HydrologyCycle/Permafrost/Obu2018/Obu2018.html#AllModels), but it still make me worried that we'll be underestimating permafrost pretty significantly with the Sturm formulation.

**Reply by olyson** [reply]
I've put in a fix for the ILAMB permafrost problem with the Sturm simulation and updated the diagnostics:
https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_SturmSnowtk_hist/lnd/_build_ILAMB2_7_CTSM51Sturm/
As expected, most of the scores for Sturm are lower than for the control.

**Reply by olyson** [reply]
I had posted an issue on the ILAMB repository regarding the permafrost issue.  This was the response:
"I have seen this problem before but have no explanation, it does not make
sense. Your solution is the right correction, I will commit. Thanks!"

**Reply by wwieder** [reply]
Glad you got this workaround to fix the issue, @olyson

**Comment by wwieder** [comment]
One more comment:  Given increases in [snow albedo](https://webext.cgd.ucar.edu/I20TR/ctsm51d145_2deg_GSWP3V1_hist/lnd/ctsm51d145_2deg_GSWP3V1_hist-ctsm51_ctsm51d116_2deg_GSWP3V1_hist/set3/set3_albedo_Polar.png) and [depth](https://webext.cgd.ucar.edu/I20TR/ctsm51d145_2deg_GSWP3V1_hist/lnd/ctsm51d145_2deg_GSWP3V1_hist-ctsm51_ctsm51d116_2deg_GSWP3V1_hist/set3/set3_landf_Russian_Arctic.png) associated with SNICAR changes, #28, we should likely start running with this on by default in testing...

**Comment by olyson** [comment]
Following on from the discussion at the CLM meeting regarding Sturm snow thermal conductivity, I've plotted the Jordon and Sturm fomulations below, with x and y axis scales same as shown in Figure 6 of Sturm1997.  I've also included a possible alternative curve (SturmEXP) where I've manipulated the Sturm quadratic equation to try to split the difference.  The Sturm expression is piecewise, with a linear expression, from a density of 0.156 g/cm3, declining to the thermal conductivity of dry air at a density of zero.  So to maintain that I've recomputed the slope of the linear expression to match the proposed alternative curve at 0.156 g/cm3.  The proposed curve fits within the 95% confidence limits shown in Figure 6 (eyeballing it).  Maybe we could run a historical with the proposed curve or something like it to see its effects, instead of trying to incorporate directly into the PPE.
The second plot shows the curves over the full range of density (air to ice).

![plot_SturmJordanTK_Figure6](https://github.com/user-attachments/assets/99db8b48-2038-4b71-9a7b-74b10b6d9c90)


![plot_SturmJordanTK_Figure6Expanded](https://github.com/user-attachments/assets/cb46a944-11dc-4f58-b242-382a68faf009)

**Reply by wwieder** [reply]
This is awesome, Keith. Thanks. I like the idea of just doing a spin up once excess ice is on with the two Sturm parameterizations to compare permafrost extent and dead arctic veg with each case. Maybe this can be coordinated with the spin up that we're already doing for the end of month code chill?

**Comment by olyson** [comment]
Below are some timeseries plots of TSOI at 3.5m for the region shown in the top plot (blue), where the ILAMB diagnostics for the 92 and 98b historicals indicate missed continuous permafrost.  The order in the second plot below is, top to bottom, 098b, 092, CMIP6-001, LE2-001.  The black lines are monthly values, while the red line indicates the annual maximum, which is what ILAMB uses.  The blue line is freezing temperature.  The last plot is just a larger version of the 098b plot.
The CMIp6-001 and LE2-001 annual max temperatures are never above freezing in any year, even at the end of the simulation, while the 098b and 092 temperatures are always above freezing.  So it seems like we'll need to drop the soil temperature by at least 2-3 degrees in 1850.  That combined with a reduction in the current positive bias in the coupled simulation over the 21st century might get us back the permafrost. 
@slevis-lmwg , maybe we can review the ILAMB permafrost python code on Wed., I'm not positive I completely understand how ILAMB is determining permafrost.

![CTSM_TSOI_timeseries_Siberia7_map](https://github.com/user-attachments/assets/e5a41d49-ad63-4fdf-8495-6df33c6b61d1)

![CTSM_TSOI_timeseries_Siberia7-2](https://github.com/user-attachments/assets/a0f8fbc5-cd44-40d0-85d6-5314282f65ea)

![CTSM_TSOI_timeseries_Siberia7-3](https://github.com/user-attachments/assets/a2e0e523-1b20-4ded-a48d-8356cef82d69)

**Comment by olyson** [comment]
Per discussion this morning, ALTMAX plots for 98b and LE2 (1995-2012).

![b e23_alpha17f BLTHIST ne30_t232 098b ANN ALTMAX polar](https://github.com/user-attachments/assets/fe2557f1-7ebf-476b-8b28-fe144e527f35)

![b e21 BHISTcmip6 f09_g17 LE2-1001 001 ANN ALTMAX polar](https://github.com/user-attachments/assets/441ef40b-421e-4dd3-be71-a4a65d0be585)

**Comment by wwieder** [comment]
Interesting PFT response here, where shrubs appear to be dead in the coupled model runs over areas where we also seem to be missing permafrost, maybe because soils are too dry?  
![image](https://github.com/user-attachments/assets/cffec7b8-2286-4705-960d-d4430de5cefb)

Thanks to @djk2120 and @linniahawkins for sharing script to look at PFT output from h1 files.

**Reply by wwieder** [reply]
Sorry, this is wrong.  I realized the script I'm using is just reading in the 'observed' LAI that's being assigned to each PFT.  I'll work on reading in an h1 file from a BGC run, but now I'm running into me memory / dask issues, unless someone has an example I can work from...?

**Reply by linniahawkins** [reply]
1) take a time mean first before regridding. 
2) only regrid the one PFT you want to look at (the script I sent loops through all 78).

**Reply by wwieder** [reply]
yeah still slow. I'll remove this thread, as it sounds like you already have something working...

**Reply by wwieder** [reply]
Here's the real data from an offline run Keith did.
![image](https://github.com/user-attachments/assets/dffa26ee-8420-4a7e-80af-5d68b9fb5416)

**Reply by linniahawkins** [reply]
Updated get_ix() function to play nice with dask, still ~25 minutes for 1 degree dataset (all PFTs):
/glade/u/home/linnia/clm5ppe/pyth/CLM6_dev_handtuning/regrid_PFToutput_example.ipynb

Here's the PFT LAI biases I'm getting when comparing CTSM52_DEV/ctsm51_ctsm51d166deadveg_1deg_CRUJRA_FLDS_ABsnoCDE_blk_A5BCD_hist/ to CLM-SP:


[ctsmdeadvegCRU-CLMSP_pftLAI.pdf](https://github.com/user-attachments/files/16576738/ctsmdeadvegCRU-CLMSP_pftLAI.pdf)

**Comment by olyson** [comment]
1850 pSASU timeseries plot of TSOI at 3.5 meters for the Sturm modification (Issue #63 SturmExp1; top plot) and the default Sturm (Issue #62 SturmCon; second plot).  (The third and fourth plots are just duplicates of the second).  Looks like SturmExp1 cools down TSOI by about 2K.  I'll put up polar plots of TLAI and TSOI once the diagnostics package finishes generating climatology.

![CTSM_TSOI_1850timeseries_Siberia7-2](https://github.com/user-attachments/assets/e71234fe-71eb-4c59-a27f-e66fd83c1f53)

**Reply by wwieder** [reply]
Thanks, Keith.  Adding polar plots of alt_max would also be helpful if you're able

**Comment by olyson** [comment]
JJA TLAI plots for 1850 pSASU for SturmExp1 and SturmCon and then a difference plot.  Some loss of TLAI but not much.
Then, ANN ALTMAX plots for the same and a difference plot.  SturmExp1 has shallower ALTMAX which is consistent with the soil temperatures being colder.

![ctsm52018_f09_xsiceON_acON_SturmExp1_pSASU JJA TLAI polar](https://github.com/user-attachments/assets/e9625382-193e-4dc7-9c82-cc72ef4a27c3)

![ctsm52018_f09_xsiceON_acON_pSASU JJA TLAI polar](https://github.com/user-attachments/assets/e537e22b-c051-47e9-aa0e-933585a8b593)

![SturmExp1-SturmCon JJA TLAI diff polar](https://github.com/user-attachments/assets/d7fab4a2-5f7c-4f1a-af66-16da1a1e4a66)

![ctsm52018_f09_xsiceON_acON_SturmExp1_pSASU ANN ALTMAX polar](https://github.com/user-attachments/assets/507e25e4-afda-4266-bea6-aa3c0bce2f23)

![ctsm52018_f09_xsiceON_acON_pSASU ANN ALTMAX polar](https://github.com/user-attachments/assets/ffc7cb88-9af5-478d-82d0-6dad3e46f055)

![SturmExp1-SturmCon ANN ALTMAX diff polar](https://github.com/user-attachments/assets/5a7f785a-02da-4731-b4ef-457115b1723b)

**Reply by wwieder** [reply]
Thanks for posting these Keith.  Do you think we should go ahead a run a historical on these (or at least the Sturm_mods experiment) so we can compare permafrost extent with ILAMB?  It's hard for me to see if we're picking up additional permafrost regions from our baseline run you posted previoiusly?
![image](https://github.com/user-attachments/assets/50ad0bf9-0f2e-45e8-95d0-05ec1be6c09d)

**Comment by olyson** [comment]
I had mentioned to Sam that I wasn't sure about the value of running historicals off of our spinups because the land-only simulations with CRUJRA-FLDS already looked decent in terms of permafrost (below is a plot from the final deadveg branch with CRUJRA-FLDS compared to Brown obs) and I figured we are more concerned with what's happening in the fully coupled simulations.  But I see your point about seeing what a 2K cooling might do in terms of extending permafrost further south.  I can start a historical.  Might be good for Sam to run his too so that the only difference will be the Sturm parameterization.  I would probably clone his as I did with the spinup cases.

![DeadVeg_CRUJRA-FLDS_Brown_Permafrost](https://github.com/user-attachments/assets/291f9874-0eb6-479d-b27a-73fcafbe6fdf)

**Comment by olyson** [comment]
The historicals noted in issues #62 and #63 are complete.  ILAMB and standard diagnostics are complete as well, but the cgd diagnostics web site and /project/diagnostics are down so I can't load them there (help request submitted).  Here is a link to some extra plots:

https://docs.google.com/presentation/d/1MPtdB2vwwxlpM6TYKU3fDLrJnpYPdYSJ/edit?usp=drive_link&ouid=104026900087612515425&rtpof=true&sd=true

Permafrost has extended further south in the SturmExp1 run, consistent with shallower ALTMAX and colder TSOI at 3.5m.  The timeseries plot of TSOI at 3.5m shows that SturmExp1 is about 2K colder in the region of interest.  TLAI is a bit reduced in JJA but vegetation is still thriving at high latitudes.  Survival percentages of all pfts are still high, with exception of a couple of high latitude types which have dropped just a few percent.  Ann Max TLAI is still high for all types.

**Reply by olyson** [reply]
Diagnostics are now posted.
